### DISCUSSION TITLE
Regarding Automation

### DISCUSSION THREAD

**Discussion by buildwithtausif** [orestbida/cookieconsent#63] [Q&A] [answered]
Hi, Dev 
I am not a so much familiar with programming but I found many websites which auto-detect the cookie in a given site and generate cookie consent on its own... So just a silly question can this automatically detect my sites cookie and according to that automatically generate a consent. @orestbida 
Regards,
Tausif

**Comment by orestbida** [comment, marked as answer]
👋 @TausifAlam007,

interesting question!

**Short answer:** 
cookie detection tools are not _super reliable_ + I would not want to maintain/keep up-to-date pre-defined cookie policies (in different languages too) 🙃.

**Long answer:**
detecting cookies is not a _trivial_ task, in fact there is no reliable tool to know all of the cookies that a website might have (as far as I know). 

Existing cookie scanner tools simply analyze the request header to determine what cookies are present inside that request. A simple technique, with a big downside: **this approach completely ignores all of the cookies loaded "on demand"** (e.g. a button which - when clicked - loads an iframe with an external service => new cookies). 

A proper cookie scanner would need to analyze the content of all the existing .js scripts, check if any of the keywords matches any of the known cookie names (_ga, _gid ... ), which is a lot of work ... heck even this would not suffice ... I think cookie scanner tools are great to help you discover what cookies are on your website, but not enough to "catch 'em all"!

As for the consent policy (text content inside the modals), it was my choice to not ship the plugin with any default policy/s; My idea - when I started developing this plugin - was for it to be a "**basic-lightweight solution which also gave the user total control over his consent policy and cookies**". 

I understand perfectly that it can be a hassle to configure it, especially if a certain website has a lot of third party services. As a matter of fact, there have been cases where I myself have driven away some users from my solution, due to how big/complex their websites were.
